### PR TITLE
Add image index

### LINKED ISSUE #4: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [x] New images that are added show up on the homepage.
- [x] These images are persisted if the browser is closed or even if the
  server is restarted.
- [x] Images are not displayed wider than 400px.
- [x] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by grantpaulson6**
Closes #4 

Add image index to homepage

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by grantpaulson6** [comment]
[ROPES - Add image index](https://trello.com/c/DgV02nqu/993-ropes-add-image-index)

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class ImagesController < ApplicationController
-  def home; end
+  def home
+    @images = Image.order(created_at: :desc)
```
Comment (jasonmally) [inline app/controllers/images_controller.rb]: Just a heads up: Image.all should list the images sorted like this by default.

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -13,13 +15,14 @@ def create
     else
       flash[:errors] = @image.errors.messages
       render :new, status: :unprocessable_entity
-      # QUESTION: would we want to redirect_to new_image_path to maintain consistent URL for client? (shows as
-      # /images after failure)
     end
   end
 
   def show
     @image = Image.find(params[:id])
```
Comment (jasonmally) [inline app/controllers/images_controller.rb]: What if someone issues a call to 'show' with no ID param passed in at all?

File: app/models/image.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 class Image < ApplicationRecord
   validates :name, :url, presence: true
+  # validates :url, url: true #(built in for Rails?)
```
Comment (jasonmally) [inline app/models/image.rb]: Yeah, this should work!

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -4,19 +4,38 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
 
-    # should we test what template is rendered?
     assert_response :success
     assert_select '#header', /Welcome/
   end
 
+  def test_home__renders_index_of_images
+    get root_path
+
+    # QUESTION: is testing for CSS class enough, or should actually count elements and compare to db?
```
Comment (jasonmally) [inline test/controllers/images_controller_test.rb]: It should be sufficient to test for CSS class here

**Pull request merged**